### PR TITLE
Flysystem v2 s3

### PR DESCRIPTION
**Pull request by Nyholm**
This adapter is working. But I cannot make sure it is working with Docker. (I have not spent too much time on it). 

The only bug I found was related to S3. Reported here: #177

### CODE REVIEW

**Comment by Nyholm** [comment]
Im not super happy with the namespace.. could that be improved?

**Comment by jderusse** [comment]
do we need to support v1?

If v3 is released would we create a new component or that component will support both version?

IMHO Flysystem\S3 works.

**Comment by Nyholm** [comment]
Hm.. Im not sure. 

I would like to be able to support both versions.

Oh, of course. We can support all verisons of flysystem with two different classes.

**Comment by jderusse** [comment]
> Oh, of course. We can support all verisons of flysystem with two different classes.

yes, that's what I mean in my second sentence

**Comment by Nyholm** [comment]
So, everything is working except #177, but that is not blocker for merging. 

The only other thing that is broken are the tests =) I dont know why docker is not helping me at all.

File: src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+     */
+    private $s3Client;
+
+    public static function setUpBeforeClass(): void
+    {
+        static::$adapterPrefix = 'travis-ci/' . bin2hex(random_bytes(10));
+
+        if (!getenv('FLYSYSTEM_AWS_S3_KEY')) {
+            // Fixme, Docker is not working
+            self::markTestSkipped('Docker image is not working.');
```
Comment (Nyholm) [inline src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php]: We dont need integration tests. They exist (or should exist) in the async-aws/s3 package.

**Comment by Nyholm** [comment]
When I test with "real" s3 it all passes. I'll merge this PR and create a package

**Pull request merged**